### PR TITLE
Taco integration

### PR DESCRIPTION
**Pull request by jbradl11**
Pull requests into CQL QDM Patient API require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

This PR integrates the work done for the TACO converter.  Regression tests come back clean, running this branch using the CQM Converter on Bonnie patients, around 100 patients fail due to invalid UCUM units, 2 others fail due to the newly added Participation model not being supported in the CQM repo. 

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] This PR is into the correct branch.
- [x] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1537
- [x] JIRA ticket links to this PR
- [x] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [x] Code coverage has not gone down and all code touched or added is covered.
    * In rare situations, this may not be possible or applicable to a PR. In those situations:
        1. Note why this could not be done or is not applicable here: Code coverage has gone down slightly due to bringing in the TACO work.
        2. Add TODOs in the code noting that it requires a test
        3. Add a JIRA task to add the test and link it here:
- [x] Automated regression test(s) pass

Branch | Front End Coverage
-- | --
master | image
taco_integration | image

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: N/A
- [x] Justification for using JIRA tests: N/A
- [x] JIRA tests have been added to sprint: N/A

**Reviewer 1:**

Name: @holmesie
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] <s>JIRA tests have been run and pass</s>
- [x] <s>You agree with the justification for use of JIRA tests or have provided input on why you disagree</s>

**Reviewer 2:**

Name: @c-monkey
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] <s>JIRA tests have been run and pass</s>
- [x] <s>You agree with the justification for use of JIRA tests or have provided input on why you disagree</s>

### CODE REVIEW

File: spec/javascripts/datatypes/assessment_spec.js.coffee [hunk 1]
```diff
@@ -3,7 +3,7 @@ describe "Assessment", ->
     it "should return a list of ids when relatedTo is called", ->
       assessmentPerformed = new CQL_QDM.AssessmentPerformed({'references': [{'referenced_id':'1234567890', 'type':'Example'},{'referenced_id':'0987654321', 'type':'Example'}]})
       expect(assessmentPerformed.relatedTo().length).toBe 2
-      referenceId = new CQL_QDM.Id('1234567890')
+      referenceId = new CQL_QDM.Id('1234567890', null, 'Example')
```
Comment (c-monkey) [inline spec/javascripts/datatypes/assessment_spec.js.coffee]: this has 3 parameters, should be only 2

File: spec/javascripts/datatypes/caregoal_spec.js.coffee [hunk 2]
```diff
@@ -1,8 +1,52 @@
...
+
+  it "should return null if no targetOutcome is specified", ->
+    careGoal = new CQL_QDM.CareGoal({})
+    expect(careGoal.targetOutcome()).toEqual null
+
   it "should return a list of ids when relatedTo is called", ->
     careGoal = new CQL_QDM.CareGoal({'references': [{'referenced_id':'1234567890', 'type':'Example'},{'referenced_id':'0987654321', 'type':'Example'}]})
     expect(careGoal.relatedTo().length).toBe 2
-    referenceId = new CQL_QDM.Id('1234567890')
+    referenceId = new CQL_QDM.Id('1234567890', null, 'Example')
```
Comment (c-monkey) [inline spec/javascripts/datatypes/caregoal_spec.js.coffee]: this has 3 parameters, should be only 2

File: spec/javascripts/datatypes/communication_spec.js.coffee [hunk 3]
```diff
@@ -3,7 +3,7 @@ describe "Communication", ->
     it "should return a list of ids when relatedTo is called", ->
       communication = new CQL_QDM.CommunicationFromPatientToProvider({'references': [{'referenced_id':'1234567890', 'type':'Example'},{'referenced_id':'0987654321', 'type':'Example'}]})
       expect(communication.relatedTo().length).toBe 2
-      referenceId = new CQL_QDM.Id('1234567890')
+      referenceId = new CQL_QDM.Id('1234567890', null, 'Example')
```
Comment (c-monkey) [inline spec/javascripts/datatypes/communication_spec.js.coffee]: this has 3 parameters, should be only 2

File: spec/javascripts/datatypes/communication_spec.js.coffee [hunk 4]
```diff
@@ -12,7 +12,7 @@ describe "Communication", ->
     it "should return a list of ids when relatedTo is called", ->
       communication = new CQL_QDM.CommunicationFromProviderToPatient({'references': [{'referenced_id':'1234567890', 'type':'Example'},{'referenced_id':'0987654321', 'type':'Example'}]})
       expect(communication.relatedTo().length).toBe 2
-      referenceId = new CQL_QDM.Id('1234567890')
+      referenceId = new CQL_QDM.Id('1234567890', null, 'Example')
```
Comment (c-monkey) [inline spec/javascripts/datatypes/communication_spec.js.coffee]: this has 3 parameters, should be only 2

File: spec/javascripts/datatypes/communication_spec.js.coffee [hunk 5]
```diff
@@ -21,6 +21,6 @@ describe "Communication", ->
     it "should return a list of ids when relatedTo is called", ->
       communication = new CQL_QDM.CommunicationFromProviderToProvider({'references': [{'referenced_id':'1234567890', 'type':'Example'},{'referenced_id':'0987654321', 'type':'Example'}]})
       expect(communication.relatedTo().length).toBe 2
-      referenceId = new CQL_QDM.Id('1234567890')
+      referenceId = new CQL_QDM.Id('1234567890', null, 'Example')
```
Comment (c-monkey) [inline spec/javascripts/datatypes/communication_spec.js.coffee]: this has 3 parameters, should be only 2

File: spec/javascripts/datatypes/id_spec.js.coffee [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
     id = new CQL_QDM.Id(null, null)
     expect(id.namingSystem).toBeNull()
 
-  it 'Should handle a blank value', ->
+  it 'should handle a blank value', ->
     id = new CQL_QDM.Id()
     expect(id.value).not.toBeDefined()
 
-  it 'Should handle a present namingSystem and value', ->
+  it 'should handle a present value, referenced_type, and type', ->
```
Comment (c-monkey) [inline spec/javascripts/datatypes/id_spec.js.coffee]: Did you temporarily add a type? Anyway, this no longer tests for type.
